src/Core/Compiler/TemplateCompiler.php

```diff
@@ -37,13 +39,49 @@ class TemplateCompiler {
 	protected $renderingContext;
 
 	/**
+	 * @var string
+	 */
+	protected $mode = self::MODE_NORMAL;
```

Comment:
The only requirement is those assignments must not call any active code, e.g. no functions, no constructors, no `constant('name')`, `isset()` or anything of that sort. But constants work regardless of how. Maybe you're thinking about the `\Vendor\Class::class` thing which is fairly recent in general?